web/main/reporting/create_reporting_views.py

```diff
@@ -0,0 +1,41 @@
+from django.db import connection
+from django.template.loader import render_to_string
+
+from ..models import Casebook
+
+
+# "Published" state matches logic in the front end, which includes casebooks under revision
+PUBLISHED_CASEBOOKS = (
+    Casebook.LifeCycle.PUBLISHED.value,
+    Casebook.LifeCycle.REVISING.value,
+)
+ALL_STATES = tuple(tag.value for tag in Casebook.LifeCycle)
+
+OLDEST_YEAR = 20  # How far back in time we'll go
+
+VIEW_LIST = (
+    "reporting_professors_with_casebooks",
+    "reporting_casebooks_from_professors",
+    "reporting_casebooks_with_multiple_collaborators",
+    "reporting_casebooks_including_source_cap",
+    "reporting_casebooks_including_source_gpo",
+)
+
+
+def create() -> None:
+    """Create a little data warehouse of prefiltered reporting-ready tables for reuse in this session.
+    This logic should match what's used in `create_search_index.sql` unless otherwise annotated."""
+
+    with connection.cursor() as cursor:
+
+        sql = render_to_string(
```

Comment:
**Question:** Does `render_to_string()` offer any protection against SQL injections in this context _(what kind of escaping does it perform)_? 

If not, my _slight_ concern here would be that future, more complex iterations of this reporting tool could use more   variables to customize `reporting.sql.j2`, which would slowly become more vulnerable?

Reply:
It does not! https://github.com/sripathikrishnan/jinjasql  does. On the other hand, this kind of async job cannot take user parameters since there's no entry point. On the third hand, it does set up a potential pattern where that could be true.

Reply:
I can definitely swap it out with just a plain text file if that would feel safer.

Reply:
> On the other hand, this kind of async job cannot take user parameters since there's no entry point. On the third hand, it does set up a potential pattern where that could be true.

Yes, I think we're on the same page 😄 

---

> I can definitely swap it out with just a plain text file if that would feel safer.

Alternatively, here's a suggestion:
- We could add a comment around `sql = render_to_string(...)`to signal that there's a slight risk here _(so future us / somebody else is mindful of what is added and how)_
- We could also add a quick format check in  `{% for view in views %}` to make sure `view` is a string matching `this_format` ?

Reply:
Eh, I'd rather not introduce complexity and readability challenges just to save a couple slightly redundant lines. I'll switch this to plain text and if we want to introduce more complex SQL templating we can do that with a purpose-built library. Thanks for raising the issue!